### PR TITLE
Fix torch.load(..., weights_only=True) for NT

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112516

Found when looking into #112509

### CODE REVIEW

**Review by soulitzer — APPROVED** [review]

File: torch/_weights_only_unpickler.py [hunk 1]
```diff
@@ -101,6 +101,7 @@ def _get_allowed_globals():
         torch._utils._rebuild_tensor_v2,
         torch._utils._rebuild_sparse_tensor,
         torch._utils._rebuild_meta_tensor_no_storage,
+        torch._utils._rebuild_nested_tensor,
```
Comment (albanD) [inline torch/_weights_only_unpickler.py:104]: Why not add torch._nested_view_from_buffer here directly?

Comment (jbschlosser) [reply torch/_weights_only_unpickler.py:104]: I tried that initially but it didn't work? It was getting caught up on the built-in getattr

Comment (albanD) [reply torch/_weights_only_unpickler.py:104]: The one for the `torch` module?

Comment (jbschlosser) [reply torch/_weights_only_unpickler.py:104]: Yes, I believe that's the one

**Pull request closed without merging**